tools/print_dependencies.py

```diff
@@ -33,4 +36,33 @@
     })
 
   deps = sorted(deps, key=lambda k: k['identifier'])
-  print json.dumps(deps, sort_keys=True, indent=2)
+
+  # Print all dependencies if a target is unspecified
+  if len(sys.argv) == 1:
+    print(json.dumps(deps, sort_keys=True, indent=2))
+    exit(0)
+
+  # Bazel target to print
+  target = sys.argv[1]
+  output = subprocess.check_output(['bazel', 'query', 'deps(%s)' % target])
+
+  repos = set()
+
+  # Gather the explicit list of repositories
+  repo_regex = re.compile('^@(.*)\/\/')
+  for line in output.split('\n'):
+    match = repo_regex.match(line)
+    if match: repos.add(match.group(1))
+
+  # Gather the build recipes repositories
+  # These are part of @envoy_deps repository ie @envoy_deps//:luajit
+  recipe_regex = re.compile('^@envoy_deps//:(\w+)$')
+  for line in output.split('\n'):
+    match = recipe_regex.match(line)
+    if not match: continue
+    key = match.group(1)
+    repo = RECIPE_MAP.TARGET_RECIPES[key]
+    if repo: repos.add(repo)
+
+  deps = filter(lambda dep: dep['identifier'] in repos, deps)
+  print(json.dumps(deps, sort_keys=True, indent=2))
```

Comment:
Could factor the print/dump into a separate function, so you have consistent formatting across both global and filtered deps.